packages/datadog-shimmer/src/shimmer.js

```diff
@@ -33,79 +52,55 @@ function wrapFunction (original, wrapper) {
   return wrapped
 }
 
-const wrapFn = function (original, delegate) {
-  throw new Error('calling `wrap()` with 2 args is deprecated. Use wrapFunction instead.')
-}
-
-// This is only used in safe mode. It's a simple state machine to track if the
-// original method was called and if it returned. We need this to determine if
-// an error was thrown by the original method, or by us. We'll use one of these
-// per call to a wrapped method.
-class CallState {
-  constructor () {
-    this.called = false
-    this.completed = false
-    this.retVal = undefined
-  }
-
-  startCall () {
-    this.called = true
-  }
-
-  endCall (retVal) {
-    this.completed = true
-    this.retVal = retVal
-  }
-}
-
 function isPromise (obj) {
-  return obj && typeof obj === 'object' && typeof obj.then === 'function'
+  return typeof obj?.then === 'function'
 }
 
 let safeMode = !!process.env.DD_INEJCTION_ENABLED
 function setSafe (value) {
   safeMode = value
 }
 
-function wrapMethod (target, name, wrapper, noAssert) {
+function wrap (target, name, wrapper, noAssert) {
+  assert(typeof name !== 'function', 'Implementor error: wrap() name should not be a function')
   if (!noAssert) {
     assertMethod(target, name)
-    assertFunction(wrapper)
+    if (typeof wrapper !== 'function') {
+      throw new Error(wrapper ? 'Target is not a function.' : 'No function provided.')
```

Comment:
```suggestion
      throw new Error(wrapper ? 'Target is not a function' : 'No function provided')
```

Reply:
I did not want to change the errors so far. So I copied it 1-to-1 as it was. I am fine either way.